test/caPutJsonLogTest.cpp

```diff
@@ -812,6 +848,31 @@ void runTests(void *arg) {
            &charArray1[0], strlen(charArray1), &charArray2[0], strlen(charArray1), "DBF_CHAR array (lso) test");
 
 
+    // Test metadata
+    std::map<std::string, std::string> metadata_test_single = {
+        {"ABCDEFGHIJKLMNOPQRS", "1234567890_1234567890"}
+    };
+    std::map<std::string, std::string> metadata_test_escape_chars = {
+        {"ABC\n##&¤67&nbsp; space&#160&&#160;\'\\n#", "CBC\n##&¤67&nbsp; space&#160&&#160;\'\\n#"}
+    };
+    std::map<std::string, std::string> metadata_test_multiple = {
+        {"One", "1"}, {"Two", "2"}, {"Three", "3"}, {"Four", "4"}, {"Five", "5"},
+        {"Six", "6"}, {"Seven", "7"}, {"Eight", "8"}, {"Nine", "9"}, {"Ten", "10"},
+    };
+
+    testMetadataHelper(metadata_test_single);
+    testMetadataHelper(metadata_test_escape_chars);
+    testMetadataHelper(metadata_test_multiple);
+
+    // 1000+ elements in metadata
+    std::map<std::string, std::string> metadata_test_big;
+    for(int i = 0; i <=1000; i++)
+    {
+        metadata_test_big.insert(std::pair<std::string, std::string>(std::to_string(i), std::to_string(i)));
+    }
+    testMetadataHelper(metadata_test_big);
+
```

Comment:
We should add a test for reinserting a key, expect that the key is there but content is replaced with the newest.

Reply:
You mean for inserting the a key which already exists and has a value, right?